File: examples/flutter_app/lib/models/device.dart
class Device {
  final String id;
  final String uid;
  final String name;
  final String type;
  final String status;
  final DateTime? lastSeen;

  Device({
    required this.id,
    required this.uid,
    required this.name,
    required this.type,
    required this.status,
    this.lastSeen,
  });

  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
      id: json['id'] ?? '',
      uid: json['device_uid'] ?? '',
      name: json['name'] ?? 'Unknown',
      type: json['type'] ?? 'unknown',
      status: json['status'] ?? 'offline',
      lastSeen: json['last_seen'] != null ? DateTime.parse(json['last_seen']) : null,
    );
  }
}

File: web/src/types/device.ts
export interface Device {
    id: string;
    name: string;
    type: string;
    device_uid: string;
    public_ip: string;
    last_seen: string;
    created_at: string;
    status: 'online' | 'offline';
    shadow_state?: Record<string, any>;
}

export interface DeviceListResponse {
    devices: Device[];
}

export interface CreateDeviceRequest {
    name: string;
    type: string;
    device_uid?: string;
}

export interface CreateDeviceResponse {
    device_id: string;
    api_key: string;
    message: string;
}

export interface Command {
    command_id: string;
    action: string;
    params: Record<string, any>;
    status?: string;
    created_at: string;
}

export interface SendCommandRequest {
    action: string;
    params?: Record<string, any>;
    expires_in?: number;
}

export interface SendCommandResponse {
    command_id: string;
    status: string;
    message: string;
    expires_at: string;
}

export interface TelemetryPoint {
    timestamp: number;
    data: Record<string, number | string | boolean>;
}

export interface TelemetryHistoryResponse {
    data: TelemetryPoint[];
}
